src/nupic/types/Types.hpp

```diff
@@ -42,8 +42,12 @@ namespace nupic {
  */
 
 /**
- * Represents a 8-bit byte.
+ * Represents a signed 8-bit byte.
+ * (note that std::byte is unsigned char.)
```

Comment:
and we do need signed?

Reply:
> nupic::Byte: and we do need signed?

No, but I have verified that it does compile with both signed and unsigned.  What I am trying to point out is that this is an integer value NOT a character.  Unfortunately we don't have a platform independent way to tell the compiler that.  Will still have problems if used with I/O on windows.

Reply:
> No, but I have verified that it does compile with both signed and unsigned. 

I recall there were some problems with Byte = std::byte ? If not, wouldn't unsigned be better for this?

Reply:
> I recall there were some problems with Byte = std::byte ?

std::byte would be ideal because this is what this type is intended for. An 8bit integer and not a character. However, it is not in the standard until C++14.   So we should leave it as it always has been, a char (which is a signed char).